### PR TITLE
[posix] remove NCP app

### PR DESCRIPTION
**Pull request by bukepo**
The POSIX NCP app(ot-ncp) is simulating of NCP with an RCP. With the latest otbr-agent, this is not needed any more. It is only used in CI tests. This commit removes the app to save maintenance efforts.

### CODE REVIEW

File: tests/toranj/build.sh [hunk 1]
```diff
@@ -174,86 +168,6 @@ case ${build_config} in
         cp -p ${top_builddir}/examples/apps/ncp/ot-ncp-ftd ${top_builddir}/examples/apps/ncp/ot-ncp-ftd-15.4-trel
         ;;
 
-    rcp)
```
Comment (abtink) [inline tests/toranj/build.sh:177]: Suggest keeping the rcp build option here.   use it a lot during development and testing (quick way to conifigure a build with rcp only).

File: tests/toranj/build.sh [hunk 2]
```diff
@@ -174,86 +168,6 @@ case ${build_config} in
...
-            --enable-coverage=${coverage} \
-            --enable-ncp \
-            --enable-radio-only \
-            --with-examples=simulation \
-            --disable-docs \
-            --enable-tests=$tests || die
-        make -j 8 || die
-        ;;
-
-    posix | posix-)
```
Comment (abtink) [inline tests/toranj/build.sh:195]: Actually I suggest keeping all the rest of configurations here in the `build.sh`. There is no harm/overhead in keeping them (even if they may not be by any CI).

It's useful tool to quickly configure a build flavor and test builds during development.

**Comment by abtink** [comment]
@bukepo , is there a big win by removing the POSIX NCP build? I don't think we have a lot of efforts/overhead in maintaining it.

My concern is that this PR seem to be removing a lot of RCP/posix test in our CI (`toranj` test and others). I prefer to keep running them on all PRs.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

Less maintenance overhead sounds better.

**Review by librasungirl — APPROVED** [review]

**Comment by librasungirl** [comment]
`wpantund+ot-ncp+rcp` seems an interim architecture when migrating from `wpantund + ncp` to `daemon + rcp` solution, unless there is outstanding use case or advantage, it may be not necessary to commit any further maintenance efforts IMHO.

File: tests/toranj/build.sh (hunk 1 above)
Comment (bukepo) [reply tests/toranj/build.sh:177]: I'm fine with keeping this part.

File: tests/toranj/build.sh (hunk 2 above)
Comment (bukepo) [reply tests/toranj/build.sh:195]: After removing the NCP app, it seems these options cannot build valid binaries for toranj tests. Are you sure want to keep these options?

**Comment by bukepo** [comment]
> @bukepo , is there a big win by removing the POSIX NCP build? I don't think we have a lot of efforts/overhead in maintaining it.
> 
> My concern is that this PR seem to be removing a lot of RCP/posix test in our CI (`toranj` test and others). I prefer to keep running them on all PRs.

I couldn't find good use cases for POSIX ncp app. It does take time to ensure CI tests passing with posix ncp simulation. I think removing it can make life of OpenThread development easier. We can add it back when one day there's a valid use case. Thoughts?

File: tests/toranj/build.sh (hunk 2 above)
Comment (abtink) [reply tests/toranj/build.sh:195]: Yes. Let's keep them. We can change them to build CLI instead of NCP.

Comment (bukepo) [reply tests/toranj/build.sh]: Reverted changes in build.sh.

**Comment by bukepo** [comment]
Tests passed, but GitHub Actions will keep waiting responses of deleted tests in PR. @abtink @jwhui

File: tests/toranj/build.sh (hunk 2 above)
Comment (abtink) [reply tests/toranj/build.sh]: I think we may need to change the script for this build flavors to actually build CLI (and not NCP anymore).
It's ok I can take care of later in separate PR.

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**